src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -42,6 +42,9 @@ class TaskKiller @Inject() (
           val activeInstances = foundInstances.filter(_.isActive)
 
           if (wipe) {
+            if (runSpec.isResident) {
+              val decommissioned = await(Future.sequence(foundInstances.map(i => instanceTracker.goalDecommissioned(i.instanceId))))
```

Comment:
yes, because this is wipe=true, so for this thing, we want to wipe the resident instance so that we can unreserve in `OfferMatcherReconciler` later (basically to wipe a reservation, we need to expunge the instance because reservation is saved on instance).

With decommissioned introduced, we might actually improve this flow in the future, see https://jira.mesosphere.com/browse/MARATHON-8295

Reply:
Got it. Thx 👍